r/R/record-batch.R

```diff
@@ -218,7 +207,12 @@ names.RecordBatch <- function(x) x$names()
   if (!missing(j)) {
     # Selecting columns is cheaper than filtering rows, so do it first.
     # That way, if we're filtering too, we have fewer arrays to filter/slice/take
-    x <- x$select(j)
+    if (is_integerish(j)) {
+      x <- x$SelectColumns(as.integer(j) - 1L)
```

Comment:
Judging from the tests, I think we want to add something like this (adapted from filter_rows in array.R):

```suggestion
      if (all(j < 0)) {
        # in R, negative j means "everything but j"
        j <- setdiff(seq_len(x$num_columns), -1 * j)
      }
      x <- x$SelectColumns(as.integer(j) - 1L)
```